Extensions/TextObject/Extension.cpp

```diff
@@ -200,6 +200,70 @@ void DeclareTextObjectExtension(gd::PlatformExtension& extension) {
       .SetFunctionName("SetColor")
       .SetIncludeFile("TextObject/TextObject.h");
 
+  obj.AddAction(
+         "ChangeGradient",
+         _("Gradient"),
+         _("Change the gradient of the text."),
+         _("Change gradient of _PARAM0_ to colors _PARAM1_ _PARAM2_ _PARAM3_ _PARAM4_ type _PARAM5_"),
+         _("Effects"),
+         "res/actions/textGradient24.png",
+         "res/actions/textGradient.png")
+
+      .AddParameter("object", _("Object"), "Text")
+      .AddParameter("color", _("First Color"))
+      .AddParameter("color", _("Second Color"))
+      .AddParameter("color", _("Third Color"))
+      .AddParameter("color", _("Fourth Color"))
+      .AddParameter("stringWithSelector", _("Gradient type"), "[\"LINEAR_VERTICAL\", \"LINEAR_HORIZONTAL\"]", false)
```

Comment:
nitpicking: Would you mind moving stringWithSelector before First Color (after Object)? This way if later we want to add a fifth color, it's possible without strange argument positioning :) (you also need to update the JS function)